modules/misc/nixgl.nix

```diff
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+
+let cfg = config.nixGL;
+in {
+  meta.maintainers = [ lib.maintainers.smona ];
+
+  options.nixGL.prefix = lib.mkOption {
+    type = lib.types.str;
+    default = "";
+    example = lib.literalExpression
+      ''"''${inputs.nixGL.packages.x86_64-linux.nixGLIntel}/bin/nixGLIntel"'';
+    description = ''
+      The nixGL command that `lib.nixGL.wrap` should wrap packages with.
+      This can be used to provide libGL access to applications on non-NixOS systems.
+
+      Wrap individual packages like so: `(config.lib.nixGL.wrap <package>)`. The returned package
+      can be used just like the original one, but will have access to libGL. If this option is empty (the default),
+      then `lib.nixGL.wrap` is a no-op. This is useful on NixOS, where the wrappers are unnecessary.
+    '';
+  };
+
+  config = {
+    lib.nixGL.wrap = # Wrap a single package with the configured nixGL wrapper
+      pkg:
+
+      if cfg.prefix == "" then
+        pkg
+      else
+      # Wrap the package's binaries with nixGL, while preserving the rest of
+      # the outputs and derivation attributes.
+        (pkg.overrideAttrs (old: {
+          name = "nixGL-${pkg.name}";
+
+          # Make sure this is false for the wrapper derivation, so nix doesn't expect
+          # a new debug output to be produced. We won't be producing any debug info
+          # for the original package.
+          separateDebugInfo = false;
+
+          buildCommand = ''
+            set -eo pipefail
+
+            ${
+            # Heavily inspired by https://stackoverflow.com/a/68523368/6259505
+            pkgs.lib.concatStringsSep "\n" (map (outputName: ''
```

Comment:
```suggestion
            lib.concatStringsSep "\n" (map (outputName: ''
```

Reply:
@vigress8 can you explain the motivation for this suggestion? I'm not super familiar with how HM handles `lib`, but I would imagine this function would be the same regardless of this change. Is that not the case?

Reply:
You're correct, but the module receives `lib` as an argument anyway, so it's more consistent to use it directly imo.